spanner/session.go

```diff
@@ -1365,6 +1483,15 @@ func (hc *healthChecker) maintainer() {
 			hc.pool.lastResetTime = now
 		}
 		hc.pool.mu.Unlock()
+
+		// task to remove or log sessions which are unexpectedly long-running
+		if now.After(hc.pool.InactiveTransactionRemovalOptions.lastExecutionTime.Add(hc.pool.executionFrequency)) {
+			if hc.pool.CloseInactiveTransactions || hc.pool.LogInactiveTransactions {
+				hc.pool.removeLongRunningSessions()
```

Comment:
* What happens if an exception gets thrown from this method? Is it handled such that it does not disrupt other things in the initpool() method?
* Also should this new task be the last task of this method?

Reply:
1. - `hc.pool.removeLongRunningSessions()` does 2 tasks. First, calls `p.getLongRunningSessionsLocked`, which iterates the list and captures the sessions that are not eligible to be long-running. Exception is not possible here
https://github.com/googleapis/google-cloud-go/pull/8177/files/f578facefc79db96da529a304394f4894a8c3529#diff-9479303281d57658abd89eb39a8b1537fb58fca056d655901309ed1839c443f3R698
   -  Secondly, it iterates over the `longRunningSessions` to destroy the sessions. For this we use existing `sh.destroy()` which does not throw any errors as a return type of this method. So, exception is not possible here too.
   https://github.com/googleapis/google-cloud-go/blob/f609b3cf831fb89c45386f81d0047560120cb3f4/spanner/session.go#L137

2. - This task cannot be moved at the end of this method, because there is a `select` which is a blocking call and waits till the window duration.
   -  I have added this task before `Grow or shrink pool` task to get some benefit though it is not really high. Ex: if our session cleaning tasks removes couple of sessions, then the grow task can increase the session to above minSessions instead of waiting until the next cycle to grow its size.

Reply:
Yes exceptions are not possible here because error is not returned, looks good to me